TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -506,7 +506,54 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
                         node.GetEnclosingProgramOrFunctionNode().SetFlag(Node.Flag.UseGlobalStorage, true);
                     }
                 }
-                
+
+                if (!isReadStorageArea && node.SymbolTable.CurrentScope == SymbolTable.Scope.Function)
+                {
+                    FunctionDeclaration functionDeclaration = node.GetParent<Nodes.FunctionDeclaration>();
+                    if (functionDeclaration != null)
+                    {
+                        // search node variable in function declaration
+                        var variables = functionDeclaration.SymbolTable.GetVariablesExplicitWithQualifiedName(
+                            area.SymbolReference != null
+                                ? area.SymbolReference.URI
+                                : new URI(area.ToString()), null);
+                        var functionVariables = variables.Select(v => v.Value).ToList();
+                        if (functionVariables.Any())
```

Comment:
You already have dataDefinitionFound or dataDefinitionPath.CurrentDataDefinition 
which reference the data declaration (outside a typedef) in the current context.

If the referenced data is outside a typedef : dataDefinitionFound  is not null.
Otherwise, you have to use dataDefinitionPath.CurrentDataDefinition 

So, there is no need to search it again using the SymbolTable.